interfaces/apparmor/backend.go

```diff
@@ -63,6 +70,138 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 	return interfaces.SecurityAppArmor
 }
 
+// Initialize prepares customized apparmor policy for snap-confine.
+func (b *Backend) Initialize() error {
+	// NOTE: It would be nice if we could also generate the profile for
+	// snap-confine executing from the core snap, right here, and not have to
+	// do this in the Setup function below. I sadly don't think this is
+	// possible because snapd must be able to install a new core and only at
+	// that moment generate it.
+	return setupSnapConfineGeneratedPolicy()
+}
+
+// setupSnapConfineGeneratedPolicyImpl inspects the system and sets up local
+// apparmor policy for snap-confine. Local policy is included by the
+// system-wide policy. If the local policy has changed then the apparmor
+// profile for snap-confine is reloaded.
+func setupSnapConfineGeneratedPolicyImpl() error {
+	// Create the local policy directory if it is not there.
+	if err := os.MkdirAll(dirs.SnapConfineAppArmorDir, 0755); err != nil {
+		return fmt.Errorf("cannot create snap-confine policy directory, %s", err)
+	}
+
+	// Check the /proc/self/exe symlink, this is needed below but we want to
+	// fail early if this fails for whatever reason.
```

Comment:
Thanks for making that clear. That sort of comment helps much when someone else goes to refactor the logic.